src/Build/BackEnd/Shared/BuildAbortedException.cs

```diff
@@ -40,9 +38,7 @@ public BuildAbortedException()
         public BuildAbortedException(string message)
             : base(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("BuildAbortedWithMessage", message))
         {
-            ResourceUtilities.FormatResourceStringStripCodeAndKeyword(out string errorCode, out _, "BuildAbortedWithMessage", message);
-
-            ErrorCode = errorCode;
+            ErrorCode = "MSB4197"; // Must match the shared resource "BuildAbortedWithMessage"
```

Comment:
Oh, I didn't even notice that you were changing this from dynamic code extraction to hard-coded; I thought you were adding a new case.

This is faster (on a build-aborted error path) but the old way maintained this invariant automatically. Think it's worth the change?

Reply:
I think so. I don't see any advantage to trying to figure it out after the fact, and I would be very suspicious of any code path that isn't on the build aborted error path doing anything here, since it's in BuildAbortedException.cs. Slightly faster and (in my opinion) clearer.

Reply:
The advantage is "if the message or code needs to change, there's exactly one place to do that". This makes it so that there are two.

Reply:
I'll add that just because we create a BuildAbortedException doesn't necessarily mean it gets thrown or, if it does, that it makes it to the user. I tried to figure out if we ever make it without the build actually having been cancelled, and I think we do in completing the build, but I am not at all confident.